packages/@jsii/go-runtime/jsii-experimental/types.go

```diff
@@ -0,0 +1,13 @@
+package jsii
+
+type Any interface{}
+
+// JsiiObj is a trait implemented by every type that can have a corresponding
+// instance within the kernel runtime. getInstanceId is needed for all generated
+// types that may be passed by reference over the wire. For classes,
+// getInstanceId returns a private property that is a jsii object instance ID so
+// that calls to the runtime referencing this object can correctly reference the
+// corresponding instance within the kernel.
+type JsiiObj interface {
```

Comment:
It would be best if we can avoid this; although I guess Go's composition model means this name will not conflict with an actual user name?

Ideally, I'd rather follow a model where all the "magic" and bookkeeping happens *strictly outside* of the managed objects. The model where we attach info on the object itself is actually somewhat problematic in Java & Go IMHO.